### ISSUE TITLE
[REST] Categories fields are not added to the results list

### ISSUE THREAD

**Issue by joseorsini** [dotCMS/core#13418]
## Expected Behavior
REST API calls should include all fields and associated items (like Categories) if the user has the right set of permissions.

## Current Behavior
If a Non-Logged in user is trying to pull contents via REST API calls, Category Fields' values are stripped from the results list, even if the Categories themselves have CMS Anonymous -> View permissions, which in theory should allow non-logged in users to view them.

## Possible Solution
Verify permissions checking on ContentletUtil.getContentPrintableMap.

## Known Workarounds
None.

## Steps to Reproduce (for bugs)

1. Run these commands:

```
curl -v -XGET "https://demo.dotcms.com/api/content/query/+ContentType:News/limit/1" > ~/local_results_non_logged_in.txt

curl -u admin@dotcms.com:admin -v -XGET "http://demo.dotcms.com/api/content/query/+ContentType:News/limit/1" > ~/local_results_logged_in.txt
```

The first curl command doesn't return the Topic Category field value (Investment Banking), despite: 

- Category has CMS Anonymous -> View permission.
<img width="1542" alt="screen shot 2018-01-17 at 11 40 06 am" src="https://user-images.githubusercontent.com/1717110/35059156-d86f8564-fb7f-11e7-99ef-f349238f3639.png">

- Parent category has CMS Anonymous -> View permission

<img width="1525" alt="screen shot 2018-01-17 at 12 14 52 pm" src="https://user-images.githubusercontent.com/1717110/35059211-0a52490e-fb80-11e7-9dde-cbe9d1f11021.png">

- News Content Type has CMS anonymous -> Edit permission.

<img width="1530" alt="screen shot 2018-01-17 at 12 14 25 pm" src="https://user-images.githubusercontent.com/1717110/35059199-fed673ac-fb7f-11e7-8968-18a5db11e15a.png">

Second curl command returns `"topic":"Investment Banking"` as part of the JSON-formatted results list. As screenshots show, this is the actual category selected for this content: 

<img width="1533" alt="screen shot 2018-01-17 at 11 39 25 am" src="https://user-images.githubusercontent.com/1717110/35059175-e9cc1b2e-fb7f-11e7-875f-b54c80bfb555.png">
<img width="1533" alt="screen shot 2018-01-17 at 11 39 43 am" src="https://user-images.githubusercontent.com/1717110/35059177-e9e21d8e-fb7f-11e7-8eb0-337586b14cec.png">

## Context
As customer stated https://dotcms.zendesk.com/agent/tickets/95020 this problem was not present on dotCMS 3.6.2 (attempted to reproduced on this dotCMS version, and works as expected). However, tests made on: 

- Customer's stack.
- demo.dotcms.com (4.2.2).
- demo3.dotcms.com (3.7.2). 

were able to get this reproduced consistently.

## Your Environment

* dotCMS version used: 4.2.x.
* Browser Name and version: any Browser.
* Operating System and version: Any OS.
* Application Server and version: Any application server.
* Java Brand and version: Any supported Java version.
* Database and version: Any supported DB engine/version.
* Application module (if apply): N/A.

**Comment by bryanboza** [comment]
Fixed, tested in the last code of release-4.3.0 and now in both cases return the topic...

**Comment by DeanGonzalez** [comment]
Passed QA: Tested on Master, both curls displaying Topic category in results
